to_done_to_html.py

```diff
@@ -0,0 +1,196 @@
+
+class ToDoneToHtmlCommand(sublime_plugin.TextCommand):
+    """ToDone to HTML Class"""
+
+    def __init__(self, view):
+        super().__init__(view)
+        self.indent_level_map = {0 : 0}
+
+    def is_visible(self):
+        """Return true if syntax is ToDone"""
+        return "ToDone" in self.view.settings().get("syntax")
+
+    def run(self, edit):
+        """Main plugin code. Collect the current selection(s) and export to HTML"""
+
+        del edit
+
+        lines = self.get_selected_text().splitlines()
+
+        processed_lines = []
+
+        last_list_level = 0
+        list_level = 0
+
+        self.parse_indents(lines)
+
+        for line in lines:
+            if line.strip() == "":
+                prefix = "<br>"
+                postfix = ""
+            elif line.strip().startswith("-") or line.strip().startswith("+"):
+                prefix = "<li>"
+                postfix = "</li>"
+            else:
+                prefix = ""
+                postfix = ""
+
+            line_nospace = line.lstrip()
+
+            indent = len(line) - len(line_nospace)
+
+            done = False
+            if line_nospace.startswith("+"):
+                done = True
+                line_nospace = line_nospace[1:]
+                # prefix = prefix + "[DONE]"
+
+            dashes, line_nospace = get_todo_level(line_nospace)
+
+            line_nospace = html.escape(line_nospace)
+
+            line_nospace = re.sub(URL_REGEX, r'<a href="\1">\1</a>', line_nospace)
+
+            prefix = prefix + "<span style=\"%s\">" % dash_to_style(dashes, indent)
+            postfix = "</span>" + postfix
+
+            if done:
+                prefix = prefix + "<span style=\"color:lightgray; background-color: white;\">"
+                postfix = "</span>" + postfix
+
+            list_level = self.indent_level_map[indent]
+
+            if list_level > last_list_level:
+                prefix = "<ul>" * (list_level - last_list_level) + "\n" + prefix
+
+            if list_level < last_list_level:
+                prefix = "</ul>" * (last_list_level - list_level) + "\n" + prefix
+
+            processed_lines.append(" " * indent + prefix + line_nospace + postfix)
+            last_list_level = list_level
+
+        for _ in range(list_level):
+            processed_lines.append("</ul>")
+
+        final_text = "<ul>\n" + "\n".join(processed_lines) + "</ul>"
+
+        show_in_browser(final_text)
+
+    def parse_indents(self, lines):
+        """Parse how indentations should correspond to HTML list levels"""
+        indent_levels = [0]
+        for line in lines:
+            spaces = len(line) - len(line.lstrip())
+            if spaces not in indent_levels:
+                indent_levels.append(spaces)
+
+        level = 0
+        for spaces in sorted(indent_levels):
+            self.indent_level_map[spaces] = level
+            level = level + 1
+
+    def get_selected_text(self):
```

Comment:
So, does the conversion process only the text that is currently selected? Or, if there isn't a selection, does it process the full document?

Reply:
Currently it only looks at selected text, but full text will be easy to add. I'll do that 👍